tests/test_scatterwidgets.py

```diff
@@ -1,58 +1,404 @@
-from typing import Any
-
 import numpy as np
-import pandas as pd
+import pyqtgraph as pg
+import pytest
 from napari_spatialdata._model import DataModel
-from napari_spatialdata._scatterwidgets import MatplotlibWidget
+from napari_spatialdata._scatterwidgets import PlotWidget
+from pyqtgraph.Qt import QtCore, QtGui, QtWidgets
+from shapely.geometry import Polygon
+
+DATA_LEN = 100
+
+
+def normalize_coordinates(coords):
+    return [tuple(map(float, coord)) for coord in coords]
+
+
+def coordinates_are_equal(coords1, coords2, tol=1e-6):
+    norm_coords1 = normalize_coordinates(coords1)
+    norm_coords2 = normalize_coordinates(coords2)
+
+    if len(norm_coords1) != len(norm_coords2):
+        return False
+
+    return all(np.allclose(c1, c2, atol=tol) for c1, c2 in zip(norm_coords1, norm_coords2))
+
+
+@pytest.fixture
+def plot_widget(qtbot):
+    """Fixture for creating a PlotWidget instance."""
+    model = DataModel()
+    widget = PlotWidget(None, model)
+    qtbot.addWidget(widget)
+    yield widget
+    widget.deleteLater()
+    qtbot.wait(50)
+
+
+def test_initialization(plot_widget):
+    """Test initialization of PlotWidget."""
+    assert plot_widget.is_widget is True
+    assert plot_widget.scatter_plot is not None
+    assert plot_widget.hovered_point is not None
+    assert plot_widget.auto_range_button is not None
+    assert plot_widget.drawing_mode_button is not None
+    assert plot_widget.rectangle_mode_button is not None
+
+
+def test_plot_data(plot_widget, prepare_cont_test_data):
+    """Test plotting data."""
+    plot_widget._onClick(*prepare_cont_test_data)
+    plot_widget.plot()
+
+    assert plot_widget.x_data is not None
+    assert plot_widget.y_data is not None
+    assert len(plot_widget.scatter.xData) == DATA_LEN
+    assert len(plot_widget.scatter.yData) == DATA_LEN
+
+
+def test_plot_data_cont(plot_widget, prepare_cont_test_data):
+    """Test building lut widget."""
+    plot_widget._onClick(*prepare_cont_test_data)
+
+    assert plot_widget.lut is not None
+    assert plot_widget.discrete_color_widget is None
+
+
+def test_plot_data_disc(plot_widget, prepare_disc_test_data):
+    """Test building disc colors widget."""
+    plot_widget._onClick(*prepare_disc_test_data)
+
+    assert plot_widget.scatter is not None
+    assert plot_widget.lut is None
+    assert plot_widget.discrete_color_widget is not None
+
+
+def test_plot_data_widget_change(plot_widget, prepare_cont_test_data, prepare_disc_test_data):
```

Comment:
The linter of VS Code says "Redefining name 'plot_widget' from outer scope (line 27) Pylint (W0621: redefined-outer-name)" in this line. If it is true that `plot_widget` is redefined and not using the function at line 27 (the one that calls `qtbot.addWidget(widget)`), this could explain why we get the segfault in the CI in the test (adding the widget to the qt bot should prevent the segfault, but if that code is not used, then the segfault is not prevented).

Reply:
(I actually don't think this is the case, but wanted to still share this)